tutorials/pretrain_tinyllama.md

```diff
@@ -0,0 +1,127 @@
+|------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Parameters                   | 1.1B                                                                                                                                                         |
+| Model Size                   | Layers: 22, Heads: 32, Query Groups: 4, Embedding Size: 2048, Intermediate Size: 5632                                                                        |
+| Sequence Length              | 2048                                                                                                                                                         |
+| Batch Size                   | 2 million tokens (2048 * 1024)                                                                                                                               |
+| Learning Rate                | 4e-4                                                                                                                                                         |
+| Learning Rate Schedule       | Cosine with 2000 warmup steps                                                                                                                                |
+| Training Data                | [SlimPajama](https://huggingface.co/datasets/cerebras/slimpajama-627b) (893 GB), [Starcoder](https://huggingface.co/datasets/bigcode/starcoderdata) (290 GB) |
+| Combined Dataset Size        | Around 950B tokens                                                                                                                                           |
+| Total Tokens During Training | 3 trillion                                                                                                                                                   |
+
+(this table was sourced from the author's [README](https://github.com/jzhang38/TinyLlama/))
+
+
+## Download datasets
+
+You can download the data using git lfs:
+
+```bash
+# Make sure you have git-lfs installed (https://git-lfs.com):
+git lfs install
+```
+
+```bash
+git clone https://huggingface.co/datasets/cerebras/slimpajama-627b data/slimpajama-raw
+git clone https://huggingface.co/datasets/bigcode/starcoderdata data/starcoderdata-raw
+```
+Around 1.2 TB of disk space is required to store both datasets.
+
+## Prepare the datasets for training
+
+In order to start pretraining lit-gpt on it, you need to read, tokenize, and write the data in binary chunks. This will leverage our `lightning.data` optimization pipeline and streaming dataset that comes with Lightning. You will need to have the tokenizer config available:
+
+```bash
+pip install huggingface_hub sentencepiece
+
+python scripts/download.py \
+   --repo_id meta-llama/Llama-2-7b-hf \
+   --access_token your_hf_token
+```
```

Comment:
Is this step required only for the tokenizer? If so, we should consider adding a `tokenizer-only` argument to `download.py`

Reply:
Yes for the tokenizer. This is currently the easiest command to get it without the hassle to direct users to manually find these files.

Reply:
added `--tokenizer_only true` to the instructions, thanks!